src/main/java/org/springframework/data/couchbase/core/ReactiveReplaceByIdOperationSupport.java

```diff
@@ -68,7 +68,7 @@ public ReactiveReplaceByIdOperationSupport(final ReactiveCouchbaseTemplate templ
 			return Mono.just(object).flatMap(o -> {
 				CouchbaseDocument converted = template.support().encodeEntity(o);
 				return template.getCollection(collection).reactive()
-						.replace(converted.getId(), converted.export(), buildReplaceOptions()).map(result -> {
+						.replace(converted.getId(), converted.export(), buildReplaceOptions(object)).map(result -> {
```

Comment:
nit: this should be o, not object, right?

Reply:
actually I wonder if the code from below is put up here and only the long cas is put into the buildReplaceOptions

Reply:
Yes. should be o.  Fixed.

Reply:
The durability also needs to go in the options.